devices/A4988/A4988 .cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.A4988
+{
+    /// <summary>
+    /// Class for controlling A4988 stepper motor driver
+    /// </summary>
+    public class A4988 : IDisposable
+    {
+        private readonly Microsteps _microsteps;
+        private readonly ushort _fullStepsPerRotation;
+        private readonly TimeSpan _sleepBetweenSteps;
+        private readonly GpioPin _stepPin;
+        private readonly GpioPin _dirPin;
+        private readonly bool _shouldDispose;
+
+        private GpioController _gpioController;
+
+        /// <summary>
+        /// Initialize a A4988 class.
+        /// </summary>
+        /// <param name="stepPin">Pin connected to STEP driver pin</param>
+        /// <param name="dirPin">Pin connected to DIR driver pin</param>
+        /// <param name="microsteps">Microsteps mode</param>
+        /// <param name="fullStepsPerRotation">Full steps per rotation</param>
+        /// <param name="sleepBetweenSteps">By changing this parameter you can set delay between steps and controll the rotation speed (less time equals faster rotation)</param>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        public A4988(byte stepPin, byte dirPin, Microsteps microsteps, ushort fullStepsPerRotation, TimeSpan sleepBetweenSteps,
+            GpioController? gpioController = null, bool shouldDispose = true)
+        {
+            _microsteps = microsteps;
+            _fullStepsPerRotation = fullStepsPerRotation;
+            _sleepBetweenSteps = sleepBetweenSteps;
+            _gpioController = gpioController ?? new GpioController();
+            _shouldDispose = shouldDispose || gpioController is null;
+            _stepPin = _gpioController.OpenPin(stepPin, PinMode.Output);
+            _dirPin = _gpioController.OpenPin(dirPin, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Controls the speed of rotation.
+        /// </summary>
+        protected virtual void SleepBetweenSteps()
+        {
+            if (_sleepBetweenSteps == TimeSpan.Zero)
+            {
+                return;
+            }
+
+            System.Threading.Thread.Sleep(_sleepBetweenSteps);
+        }
+
+        /// <summary>
+        /// Rotates a stepper motor.
+        /// </summary>
+        /// <param name="degree">Rotation degree</param>
+        /// <param name="rotation">Direction of rotation</param>
+        public virtual void Rotate(ushort degree, bool rotation)
```

Comment:
why do you need a rotation? Just replace the degree ushort by a short. If negative, it's one way, if positive, it's the other way.
And then why taking just a short, why not directly a double?
And finaly, your rotation does not seems to be used here anyway

Reply:
You are absolutely right, it's much simpler with double and rotating based on negative/positive values. Changed to only one parameter.
BTW the rotation was used in:
`_dirPin.Write(rotation> 0 ? PinValue.High : PinValue.Low);`
now it is 
`_dirPin.Write(degree > 0 ? PinValue.High : PinValue.Low);`